### PR TITLE
HVG-486 validation

### PR DESCRIPTION
**Pull request by MathiasOhrstrom**
Item from design: []

### Checklist

- [x] Functionality is covered by an integration test
- [x] Functionality is accessible in engineering mode

### CODE REVIEW

File: testdata/src/main/java/com/hedvig/app/testdata/feature/embark/builders/TextActionBuilder.kt [hunk 1]
```diff
@@ -5,20 +5,31 @@ import com.hedvig.android.owldroid.graphql.EmbarkStoryQuery
...
-                ),
-                placeholder = placeholder
+                )
             )
         ),
         asEmbarkTextActionSet = null,
     )
+
+    companion object {
+        const val PERSONAL_NUMBER = "PersonalNumber"
```
Comment (hedvigoscar) [inline testdata/src/main/java/com/hedvig/app/testdata/feature/embark/builders/TextActionBuilder.kt:28]: Are we being delivered these masks as `String`s? 😞 

We might want to update the API in that case, these are really `enum`s in disguise 😬

**Review by hedvigoscar — CHANGES_REQUESTED** [review]
Looks pretty good! A few things to address, however. 😉

File: testdata/src/main/java/com/hedvig/app/testdata/feature/embark/builders/TextActionBuilder.kt (hunk 1 above)
Comment (MathiasOhrstrom) [reply testdata/src/main/java/com/hedvig/app/testdata/feature/embark/builders/TextActionBuilder.kt:28]: I asked Palmen if we should have these as enums instead but i think there was a reason they are strings but i dont remember why

Comment (hedvigoscar) [reply testdata/src/main/java/com/hedvig/app/testdata/feature/embark/builders/TextActionBuilder.kt:28]: Let's get some first-hand information! @palmenhq

Comment (palmenhq) [reply testdata/src/main/java/com/hedvig/app/testdata/feature/embark/builders/TextActionBuilder.kt:28]: We don't really have enum functionality in XML (which backs embark) so i don't think it makes sense to mask them as enums. If we do we would be to explode the server if a content editor adds something different than a valid enum value and i think it's just better (also for forward/backward compatibility reasons) to let them be strings and ignore any "invalid" value.

Comment (hedvigoscar) [reply testdata/src/main/java/com/hedvig/app/testdata/feature/embark/builders/TextActionBuilder.kt:28]: Well, the set of valid values on this string is in effect an enum, right? GraphQL-code gen generally takes into account that the valid enum values might change in the future and hence generates an '__UNKNOWN'-case for the enum.

To me, this seems like an example of a Stringly typed API, which I'm not in favour of 😕

Comment (palmenhq) [reply testdata/src/main/java/com/hedvig/app/testdata/feature/embark/builders/TextActionBuilder.kt:28]: Right. Although i don't fully agree i can't really say i have a strong opinion either :) feel free to open a pr in HedvigInsurance/embark or add it to the web backlog if you feel strongly for enums. I don't think we can put this as high prio from a web perspective with project bolt ongoing unfortunately :/

**Review by hedvigoscar — APPROVED** [review]

**Pull request merged**